src/MudBlazor.Analyzers/MudComponentUnknownParametersAnalyzer.cs

```diff
@@ -0,0 +1,226 @@
+                    {
+                        if (expressionStatement.Operation is IInvocationOperation invocation)
+                        {
+                            var targetMethod = invocation.TargetMethod;
+
+                            if (targetMethod.ContainingType.IsEqualTo(RenderTreeBuilderSymbol))
+                            {
+                                if (string.Equals(targetMethod.Name, "OpenComponent", StringComparison.Ordinal) && targetMethod.TypeArguments.Length == 1)
+                                {
+                                    var componentType = targetMethod.TypeArguments[0];
+                                    if (componentType.IsOrInheritFrom(MudComponentBaseType))
+                                    {
+                                        currentComponent = componentType;
+                                        currentComponentDescriptor = _componentDescriptors.GetOrAdd(currentComponent, ComponentDescriptor.GetComponentDescriptor(componentType, ParameterSymbol));
+                                    }
+                                }
+                                else if (string.Equals(targetMethod.Name, "CloseComponent", StringComparison.Ordinal))
+                                {
+                                    currentComponent = null;
+                                    currentComponentDescriptor = null;
+                                }
+                                else if (currentComponent is not null && targetMethod.Name is "AddAttribute" or "AddComponentParameter")
+                                {
+                                    if (targetMethod.Parameters.Length >= 2 && targetMethod.Parameters[1].Type.IsString())
+                                    {
+                                        var pName = invocation.Arguments[1].Value.ConstantValue;
+
+                                        if (pName.HasValue && pName.Value is string parameterName)
+                                            ValidateAttribute(context, invocation, currentComponentDescriptor, currentComponent, parameterName, className);
+                                    }
+                                }
+                            }
+                            else if (string.Equals(targetMethod.ContainingType.MetadataName, "TypeInference", StringComparison.Ordinal))
+                            {
+                                var methods = context.FilterTree.GetRoot().DescendantNodes().OfType<MethodDeclarationSyntax>();
+                                var method = methods?.Where(x => x.Identifier.ValueText == targetMethod.MetadataName).SingleOrDefault();
+
+                                if (method is not null)
+                                {
+                                    var op = context.Compilation.GetSemanticModel(method.SyntaxTree).GetOperation(method);
+                                    if (op != null)
```

Comment:
Mostly all other nullable checks uses the pattern matching
```suggestion
                                    if (op is not null)
```